### PR TITLE
[Payment][PaymentBundle] Add immutable code to Payment entities

### PR DESCRIPTION
**Pull request by tuka217**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets |  #3639,  #3617
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: features/backend/payment_methods.feature [hunk 1]
```diff
@@ -128,3 +132,15 @@ Feature: Payment methods
           And I click "delete" from the confirmation modal
          Then I should still be on the payment method index page
           And I should not see payment method with name "PayPal" in that list
+
+    Scenario: Cannot update shipping method code
```
Comment (michalmarcinkowski) [inline features/backend/payment_methods.feature]: I suppose you meant `payment method`

**Comment by pjedrzejewski** [comment]
Please rebase.

**Comment by pjedrzejewski** [comment]
@tuka217 Could you reduce this PR to a single commit? If rebase did not help then you maybe need to cherry-pick the last commit - it will be easier. I see 4 commits which should not be part of this PR.

File: src/Sylius/Bundle/PaymentBundle/spec/Form/Type/PaymentMethodTypeSpec.php [hunk 2]
```diff
@@ -73,6 +73,11 @@ function it_builds_form_with_proper_fields(
             ->shouldBeCalled()
         ;
 
+        $builder
+            ->addEventSubscriber(Argument::type('Sylius\Bundle\ResourceBundle\Form\EventSubscriber\AddCodeFormSubscriber'))
```
Comment (stloyd) [inline src/Sylius/Bundle/PaymentBundle/spec/Form/Type/PaymentMethodTypeSpec.php]:
```php
Argument::type(AddCodeFormSubscriber::class)

Comment (pjedrzejewski) [inline src/Sylius/Bundle/PaymentBundle/spec/Form/Type/PaymentMethodTypeSpec.php]: :+1:

**Comment by pjedrzejewski** [comment]
Thank you again Ania! :)

**Pull request merged**